### ISSUE TITLE
Custom loss function error

### ISSUE THREAD

**Issue by Alexthegamerr** [keras-team/keras#9752]
I am trying to implement my own custom function howherer using
```
def custom_loss(y_true, y_pred):
	x_ = ( y_true[2] - y_pred[2]) ** 2
	y_ = ( y_true[3] - y_pred[3]) ** 2
	w_ = ( y_true[4] - y_pred[4]) ** 2
	h_ = ( y_true[5] - y_pred[5]) ** 2 
	result = x_ + y_ + w_ + h_
	print(result)
	return result
```
and 

`model.compile(loss=custom_loss, optimizer=keras.optimizers.Adam(),metrics=['accuracy'])`

throws error:

InvalidArgumentError (see above for traceback): Incompatible shapes: [6] vs. [5]

It works when i use some keras loss funtion. How can i fix this?

**Comment by Dref360** [comment]
Hi!
This issue isn't related to a bug/enhancement/feature request or other accepted types of issue.

To ask questions, please see the following resources : 
* https://keras.io/
* https://keras.io/#support
* https://gitter.im/Keras-io/Lobby

Thanks!

If you think I made an error, please re-open this issue.
